### PR TITLE
add documents & update json to remove JSON namespace

### PR DESCRIPTION
**Pull request by Yexiaoxing**

### CODE REVIEW

**Comment by Yexiaoxing** [comment]
we will be able to visit the document with https://cuhkshenzhen.github.io/CUHKSZLib/

**Comment by Yexiaoxing** [comment]
Need to update Travis CI to enable auto-updating documents. https://gist.github.com/vidavidorra/548ffbcdae99d752da02

**Comment by alesiong** [comment]
Document should in `gh-pages` branch, not `master`

**Comment by alesiong** [comment]
I cannot see the real changes due to the tons of document

**Comment by alesiong** [comment]
I've read that gist, but now the log of travis-ci is down, so I cannot test it.

File: src/include/string_utils/StringCast.h [hunk 1]
```diff
@@ -15,6 +16,7 @@ class stringCast {
       strStream >> casted;
     }
     catch (std::exception &e) {
+      std::cout << "not able to cast: " << e.what() << std::endl;
```
Comment (alesiong) [inline src/include/string_utils/StringCast.h]: should be `cerr`

Comment (Yexiaoxing) [inline src/include/string_utils/StringCast.h]: you're right

**Pull request merged**